src/rustdoc_gen.rs

```diff
@@ -123,11 +182,48 @@ pub(crate) enum CrateType<'a> {
     },
 }
 
+#[derive(Debug, Clone, Hash)]
+pub(crate) struct FeatureConfig {
+    pub(crate) base_features: BaseFeatures,
+    pub(crate) extra_features: Vec<String>,
+    pub(crate) ignore_non_existing: bool,
+}
+
+#[derive(Debug, Clone, Hash)]
+pub(crate) enum BaseFeatures {
+    All,
+    Default,
+    Heuristic,
+    None,
+}
+
+impl FeatureConfig {
+    pub(crate) fn default(is_baseline: bool) -> Self {
+        // The default behaviour is the heuristic approach.
+        Self {
+            base_features: BaseFeatures::Heuristic,
+            extra_features: Vec::new(),
+            ignore_non_existing: is_baseline,
+        }
+    }
+
+    /// Unique identifier, used to mark generated rustdoc files.
+    fn make_identifier(&self) -> String {
+        if matches!(self.base_features, BaseFeatures::Heuristic) && self.extra_features.is_empty() {
+            // If using the default config, return empty identifier
+            return String::new();
+        }
+        let mut s = std::collections::hash_map::DefaultHasher::new();
+        std::hash::Hash::hash(&self, &mut s);
```

Comment:
I don't think this will work. The default hasher is randomized, and even if it weren't, it's intentionally unstable and its outputs may change between Rust versions. It isn't suitable for use as part of the cache key.

Separately, consider adding a test case that ensures that a specific feature configuration produces a specific hash value. Such a test case would have caught the problem I'm pointing out here.